src/Collection/DefaultIterator.php

```diff
@@ -127,7 +127,7 @@ public function __destruct()
     public function close()
     {
         $isClosed = false;
-        if (is_resource($this->resultId)) {
+        if (is_resource($this->resultId) || is_object($this->resultId)) {
```

Comment:
Could we have that static method added to `Laminas\Ldap\Ldap` please? Imho it makes no sense to have this separated from the class everyone is importing with `use` atm. That class handles the connection resource should have all methods available that are required to work with that "resource". 

Also having this as a static method is a big plus, because I would then start using it my code base. And could we name it `isConnectionHandle()`, since there multiple "handles" coming with the ldap extension.

And `is_object()` seems weird to me, since I could pass any object and this would return true (same with `is_resource()`). Is there a reason not to test if object is an instance of `LDAP\Connection`?

Thanks.

Reply:
I'm fine with adding that to the `LDAP` class (even though it's by now kind of a god-class).

Though I'd opt against the `Connection` naming as that actually is *not*  a connection! Even though it is emitted by `ldap_connect` - which is also grossly misnamed as it does not connect to the LDAP-Server. 

The check for `Ldap\Connection` was indeed a though I mentioned above. But as the current code used `is_object` and I merely wanted to move that into it's own method I kept that. But refactoring that once it is in it's own method is easy.

Reply:
Or use two different methods: `Ldap::isLdapHandle()` and `Ldap::isResultHandle()`?

Beware that I rebased the code and force-pushed to your branch!!  So your local copy might be out of sync!

Reply:
In general I'm fine with the version-compare :+1: